# **MÓDULO 19**
# Exercício: Estatística Aplicada

**Efetividade de Duas Estratégias de Ensino**

Imagine que uma escola esteja avaliando a eficácia de duas estratégias de ensino de matemática para alunos do ensino médio. Eles querem determinar se há uma diferença significativa no desempenho médio dos alunos entre as duas estratégias.

# **Hipóteses:**

* Hipótese nula (H0): A média das notas dos alunos na estratégia A é igual à média das notas dos alunos na estratégia B.
* Hipótese alternativa (H1): A média das notas na Estratégia B é maior do que a média das notas na Estratégia A.

# **Dados:**

* Amostra da Estratégia A: Notas de 50 alunos que receberam a Estratégia A.
* Amostra da Estratégia B: Notas de 50 alunos que receberam a Estratégia B.

Usaremos um teste Z para comparar as médias das notas entre as duas amostras.

Se o p-valor do teste Z for menor que um nível de significância pré-determinado (por exemplo, α = 0.05), rejeitamos a hipótese nula e concluímos que há uma diferença significativa nas médias das notas entre as duas estratégias de ensino.

In [7]:
import numpy as np
import matplotlib.pyplot as plt
from scipy import stats

Os dados são criados a seguir:

In [10]:
# Definindo médias e desvios padrão para as notas nas duas estratégias
media_estrategia_A = 70
desvio_padrao_estrategia_A = 10

media_estrategia_B = 75
desvio_padrao_estrategia_B = 12

# Gerando as amostras de notas para cada estratégia de ensino da nossa base
np.random.seed(0)  # Para garantir a reprodutibilidade dos resultados
amostra_estrategia_A = np.random.normal(loc=media_estrategia_A, scale=desvio_padrao_estrategia_A, size=50)
amostra_estrategia_B = np.random.normal(loc=media_estrategia_B, scale=desvio_padrao_estrategia_B, size=50)

print("Notas da Estratégia A:", amostra_estrategia_A[:5])
print("Notas da Estratégia B:", amostra_estrategia_B[:5])

Notas da Estratégia A: [87.64052346 74.00157208 79.78737984 92.40893199 88.6755799 ]
Notas da Estratégia B: [64.25440127 79.64282997 68.87033835 60.83241379 74.66181326]


# 1) De acordo com as informações analisadas o nosso teste é unilateral á direita, esquerda ou bicaudal? Justifique.




### Questão 1

O teste é **unilateral à direita**.

A hipótese alternativa afirma que a média da Estratégia B é
**maior** que a da Estratégia A — não apenas diferente, mas
especificamente maior. Isso define a região crítica no lado
direito da distribuição.

Se fosse "diferente de" (sem direção), seria bicaudal.
Se fosse "menor que", seria unilateral à esquerda.

# 2) Calcule as médias para as duas amostragens e as variâncias. Quais insights você pode retirar comparando os dados?

In [15]:
# Médias e variâncias
media_A = np.mean(amostra_estrategia_A)
media_B = np.mean(amostra_estrategia_B)
variancia_A = np.var(amostra_estrategia_A)
variancia_B = np.var(amostra_estrategia_B)

print(f"Estratégia A — Média: {media_A:.2f} | Variância: {variancia_A:.2f}")
print(f"Estratégia B — Média: {media_B:.2f} | Variância: {variancia_B:.2f}")

Estratégia A — Média: 71.41 | Variância: 126.68
Estratégia B — Média: 74.75 | Variância: 108.26


### Questão 2

**Médias:**
- Estratégia A: 71.41
- Estratégia B: 74.75

A Estratégia B apresenta média superior em 3.34 pontos.
Isso é coerente com o que esperávamos pela hipótese alternativa,
porém ainda precisamos do teste estatístico para confirmar se
essa diferença é significativa ou apenas fruto do acaso.

**Variâncias:**
- Estratégia A: 126.68
- Estratégia B: 108.26

A Estratégia A tem maior variância, ou seja, as notas dos alunos
são mais espalhadas. A Estratégia B além de ter média maior,
tem resultados mais consistentes entre os alunos.

# 3) Imprima os resultados da estatística do teste Z, p value e indique se rejeitaremos ou não a hipótese nula. Justifique sua resposta.

In [18]:
from statsmodels.stats.weightstats import ztest

estatistica_z, p_valor = ztest(amostra_estrategia_B, amostra_estrategia_A, alternative='larger')

print(f"Estatística Z: {estatistica_z:.4f}")
print(f"P-valor: {p_valor:.4f}")

if p_valor < 0.05:
    print("\nResultado: Rejeitamos a hipótese nula.")
    print("Há evidência estatística de que a Estratégia B é superior.")
else:
    print("\nResultado: Não rejeitamos a hipótese nula.")
    print("Não há evidência suficiente de diferença entre as estratégias.")

Estatística Z: 1.5267
P-valor: 0.0634

Resultado: Não rejeitamos a hipótese nula.
Não há evidência suficiente de diferença entre as estratégias.


# 4) Para finalizar monte o gráfico da distribuição da estatística do nosso teste Z e explique o que pode ser observado através dele.

In [ ]:
#Seu código aqui